### PR TITLE
Support HTTP tokens for IMDSv2

### PR DESCRIPTION
**Pull request by max-lobur**
## what
* Similar to [terraform-aws-ec2-instance](https://github.com/cloudposse/terraform-aws-ec2-instance/blob/master/docs/terraform.md#input_metadata_http_put_response_hop_limit) this should support http_* options for the metadata
* `_required` flag does not actually follow [best practices](https://docs.cloudposse.com/reference/best-practices/terraform-best-practices/#use-positive-variable-names-to-avoid-double-negatives) but this is to match the existing ec2 module (see above) 
* `http_endpoint` is not present in the provider so I did not add it.

## why
* We want to enforce mandatory IMDSv2 HTTP tokens control for Spotinst pool similar to regular pools.

## references
* https://registry.terraform.io/providers/spotinst/spotinst/latest/docs/resources/ocean_aws#instance_metadata_options
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#metadata-options

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -20,6 +20,18 @@ variable "instance_profile" {
   default     = null
 }
 
+variable "metadata_http_tokens_required" {
```
Comment (nitrocode) [inline variables.tf:23]: We usually use an `_enabled` naming convention

```suggestion
variable "metadata_http_tokens_enabled" {
```

File: main.tf [hunk 2]
```diff
@@ -52,6 +52,11 @@ resource "spotinst_ocean_aws" "this" {
     }
   }
 
+  metadata_options {
+    http_tokens                 = var.metadata_http_tokens_required ? "required" : "optional"
```
Comment (nitrocode) [inline main.tf:56]:
```suggestion
    http_tokens                 = var.metadata_http_tokens_enabled ? "required" : "optional"
```

File: variables.tf (hunk 1 above)
Comment (max-lobur) [reply variables.tf:23]: See https://github.com/cloudposse/terraform-aws-ec2-instance/blob/master/docs/terraform.md#input_metadata_http_tokens_required

I want to be consistent in two modules. Let me know if you want to fix both then

Comment (max-lobur) [reply variables.tf:23]: I think initially it was because the resulting option is "required", so looks like just an oversight in the ec2 module

Comment (nitrocode) [reply variables.tf:23]: Ah ok that makes sense then I wouldn't change it. Let's keep it consistent.

**Review by nitrocode — DISMISSED** [review]

File: variables.tf (hunk 1 above)
Comment (max-lobur) [reply variables.tf:23]: Decided to not change to keep both consistent and backward compatible

**Review by nitrocode — DISMISSED** [review]

**Review by nitrocode — APPROVED** [review]

**Pull request merged**